drivers/sensor/asahi_kasei/akm09918c/akm09918c.c

```diff
@@ -66,9 +75,9 @@ int akm09918c_sample_fetch_helper(const struct device *dev, enum sensor_channel
 		return -EBUSY;
 	}
 
-	*x = sys_le16_to_cpu(buf[1] | (buf[2] << 8));
-	*y = sys_le16_to_cpu(buf[3] | (buf[4] << 8));
-	*z = sys_le16_to_cpu(buf[5] | (buf[6] << 8));
+	*x = buf[1] | (buf[2] << 8);
+	*y = buf[3] | (buf[4] << 8);
+	*z = buf[5] | (buf[6] << 8);
```

Comment:
This is required for Big Endian arch's. Please discard change.

Reply:
> From my Perspective doing the conversion this way (buf[1] | (buf[2] << 8)) should already taken care about endianness because the shift operator works at the level of the logical value, not how it is ordered in memory.
> << shifts always to the the MSB (even across byte boarders) disregarding the endianness. << is also an arithmetic operation.
> 
> Endianness only affects how values are stored in memory (the order of bytes), but it doesn’t affect how the CPU performs bitwise operations on the value itself.
> The CPU’s shift operation takes place on the entire bit pattern, treating it as a single number, not as a collection of bytes in memory.
>
> Do you have a BE SoC to test it?

Not sure I understand: the code is mapping an array of bytes to an `int16_t` based on  your CPU endianness. Doing the same bit-shifting on Little-endian and Big-endian will give you different results.

Reply:
Update: I went and looked an old test-code I had for this use-case and ran it on BE and LE. Here's the results:

Source code:
``` c
#include <stdio.h>
#include <stdint.h>

int main(void)
{
    printf("Exercise: Converting little-endian value in big-endian system!\n");

{
    uint8_t array[] = {0x12, 0x34};
    uint16_t *val = (uint16_t *)array;

    uint16_t num = array[0] | (array[1] << 8);

    printf("Array interpreted as integer: 0x%04X = %d\n", *val, *val);
    printf("Integer after conversion: 0x%04X = %d\n", num, num);
}

    return 0;
}
```

``` console
ubieda@ubieda-ubuntu:~/programming/c_programming/endianness/src$ mips-linux-gnu-gcc-10 main.c -o main_be.o && qemu-mips -L /usr/mips-linux-gnu/ ./main_be.o
Exercise: Converting little-endian value in big-endian system!
Array interpreted as integer: 0x1234 = 4660
Integer after conversion: 0x3412 = 13330
ubieda@ubieda-ubuntu:~/programming/c_programming/endianness/src$ gcc main.c -o main.o && ./main.o
Exercise: Converting little-endian value in big-endian system!
Array interpreted as integer: 0x3412 = 13330
Integer after conversion: 0x3412 = 13330
```

It appears this is indeed needed. I suggest putting it on a separate patch or creating another PR with proper commenting.